### PR TITLE
Add HorribleSubs

### PR DESCRIPTION
**Pull request by AbdullahM0hamed**

### CODE REVIEW

**Comment by AbdullahM0hamed** [comment]
434: Requested PR Unavailable

**Comment by ArjixGamer** [comment]
i wanted to add this provider but it was a bother to make it episodic and to also paginate the results

File: anime_downloader/sites/horriblesubs.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        ]
+
+    def _scrape_episodes(self):
+        show_id = re.search("var\shs_showid\s=\s(.*?);", helpers.get(self.url).text).group(1)
+
+        next_id = 1
+        episodes = []
+
+        while True:
+            resp = helpers.get('https://horriblesubs.info/api.php', params = {'method': 'getshows', 'type': 'show', 'showid': show_id, 'nextid': next_id})
```
Comment (ArjixGamer) [inline anime_downloader/sites/horriblesubs.py:32]: this looks exactly just like how HorribleDownloader works

**Comment by Blatzar** [comment]
Errors on `anime -ll DEBUG dl "neverland" -e 1 -u --provider horriblesubs`
```

Traceback (most recent call last):
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/bin/anime", line 11, in <module>
    load_entry_point('anime-downloader==4.3.0', 'console_scripts', 'anime')()
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/anime_downloader-4.3.0-py3.8.egg/anime_downloader/cli.py", line 53, in main
    cli()
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/click-7.1.2-py3.8.egg/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/anime_downloader-4.3.0-py3.8.egg/anime_downloader/commands/dl.py", line 95, in command
    anime = cls(anime_url, quality=quality,
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/anime_downloader-4.3.0-py3.8.egg/anime_downloader/sites/anime.py", line 84, in __init__
    self._episode_urls = self.get_data()
  File "/tmp/anime_downloader_HorribleSubs_6279eab0-c5ea-11ea-b727-90324b7a94df/lib/python3.8/site-packages/anime_downloader-4.3.0-py3.8.egg/anime_downloader/sites/anime.py", line 171, in get_data
    if not isinstance(self._episode_urls[0], tuple):
IndexError: list index out of range
```

**Comment by vn-ki** [comment]
We already have nyaa. This isnt much different, right?

Also, conflicts in README.

**Comment by Blatzar** [comment]
Yeah, but lead to a lot of discussion hence why I was hesitant. Guess it's ok :thumbsup:

**Pull request merged**